tests/test_2024_10_08.py

```diff
@@ -0,0 +1,303 @@
+    ), "Old address is incorrect"
+
+    # L1 Bridge doesn't have version before update
+    try:
+        l1_token_bridge.getContractVersion()
+    except VirtualMachineError:
+        pass
+
+    # Upgrade LidoLocator implementation
+    assert lido_locator_proxy.proxy__getImplementation() == LIDO_LOCATOR_IMPL, "Old address is incorrect"
+
+    # Multisig hasn't been assigned as deposit enabler
+    assert not l1_token_bridge.hasRole(DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG)
+
+
+def check_post_upgrade_state(vote_tx):
+    l1_token_bridge_proxy = interface.OssifiableProxy(L1_OPTIMISM_TOKENS_BRIDGE)
+    l1_token_bridge = interface.L1LidoTokensBridge(L1_OPTIMISM_TOKENS_BRIDGE)
+    lido_locator_proxy = interface.OssifiableProxy(LIDO_LOCATOR)
+
+    # L1 Bridge has new implementation
+    assert (
+        l1_token_bridge_proxy.proxy__getImplementation() == L1_OPTIMISM_TOKENS_BRIDGE_IMPL_NEW
+    ), "New address is incorrect"
+
+    # update L1 Bridge to 2 version
+    assert l1_token_bridge.getContractVersion() == 2
+
+    # check deposits are still paused
+    assert not l1_token_bridge.isDepositsEnabled()
+
+    # LidoLocator has new implementation
+    assert (
+        lido_locator_proxy.proxy__getImplementation() == LIDO_LOCATOR_IMPL_NEW
+    ), "New LidoLocator address is incorrect"
+
+    # Multisig has been assigned as deposit enabler
+    assert l1_token_bridge.hasRole(DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG)
+
+    interface.OptimismPortal2(L1_OPTIMISM_PORTAL_2)  # load OptimismPortal2 contract ABI to see events
+
+    evs = group_voting_events(vote_tx)
+
+    validate_contract_upgrade(evs[0], L1_OPTIMISM_TOKENS_BRIDGE_IMPL_NEW)
+    validate_contract_finalization(evs[1], 2)
+    validate_contract_upgrade(evs[2], LIDO_LOCATOR_IMPL_NEW)
+    validate_contract_role_granting(evs[3], DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG, AGENT)
+    validate_optimism_upgrade_call(
+        evs[4],
+        L2_OPTIMISM_GOVERNANCE_EXECUTOR,
+        AGENT,
+        L2_OPTIMISM_TOKENS_BRIDGE,
+        L2_OPTIMISM_TOKENS_BRIDGE_IMPL_NEW,
+        L2_OPTIMISM_WSTETH_TOKEN,
+        L2_OPTIMISM_WSTETH_TOKEN_IMPL_NEW,
+    )
+
+
+def validate_contract_upgrade(events: EventDict, implementation: str):
+    _events_chain = ["LogScriptCall", "LogScriptCall", "Upgraded", "ScriptResult"]
+    validate_events_chain([e.name for e in events], _events_chain)
+    assert events.count("Upgraded") == 1
+    assert events["Upgraded"]["implementation"] == implementation, "Wrong implementation"
+
+
+def validate_contract_finalization(events: EventDict, version: int):
+    _events_chain = ["LogScriptCall", "LogScriptCall", "ContractVersionSet", "ScriptResult"]
+    validate_events_chain([e.name for e in events], _events_chain)
+    assert events.count("ContractVersionSet") == 1
+    assert events["ContractVersionSet"]["version"] == version
+
+
+def validate_contract_role_granting(events: EventDict, role: str, account: str, sender: str):
```

Comment:
there's 'validate_grant_role_event' method in permissions.py
But I think we can leave as is